plugins/techdocs-react/src/context.tsx

```diff
@@ -144,7 +152,10 @@ export const TechDocsReaderPageProvider = memo(
     );
   },
   (prevProps, nextProps) => {
-    return areEntityRefsEqual(prevProps.entityRef, nextProps.entityRef);
+    return (
+      areEntityRefsEqual(prevProps.entityRef, nextProps.entityRef) &&
+      arePathsEqual(prevProps.path, nextProps.path)
```

Comment:
Maybe worth adding a test for this change too, to make sure the path is what we expect.